cime_config/acme/machines/template.case.run

```diff
@@ -1,416 +1,327 @@
-            {
-                appendCaseStatus("Model did not complete - see $logfile");
-                exit(-1);
-            }
-
-        }
-        else
-        {
-            appendCaseStatus("Model did not complete 2 - no cpl.log file, $cpllogfile,  corresponding to most recent model log: $logfile");
-            exit(-1);
-        }
-    }
-
-    chdir($rundir) or $logger->logdie ("could not cd to $rundir");
-
-    if(defined $config{'LOGDIR'} && (length($config{'LOGDIR'}) > 0))
-    {
-        if(! -e $config{'LOGDIR'})
-        {
-            mkpath($config{'LOGDIR'}) or $logger->warn ("Cannot make log directory $config{'LOGDIR'}, $!");
-        }
-
-        $logger->debug("lid: $LID");
-	foreach my $comp (qw(atm cpl ocn wav glc ice rof lnd acme)){
-	    disposeLog($config{RUNDIR},$config{CASEROOT}, $comp ,$LID,$config{LOGDIR});
-	}
-    }
-
-    # Run the getTiming script if desired
-    my $owd = getcwd();
-    chdir $config{'CASEROOT'};
-    if($config{'CHECK_TIMING'} eq 'TRUE')
-    {
-        my $timingDir = $config{'CASEROOT'} . "/timing";
-        mkpath $timingDir if(! -e $timingDir);
-        $logger->info( "running timing script..");
-        system(" $config{'CASETOOLS'}/getTiming -lid $LID ");
-        $logger->info( "gzipping timing stats..");
-        qx( gzip timing/acme_timing_stats.$LID );
-    }
-    chdir $owd;
-
-    # Save the timing files if desired
-    if($config{'SAVE_TIMING'} eq 'TRUE')
-    {
-        move( "$config{'RUNDIR'}/timing",  "$config{'RUNDIR'}/timing.$LID");
-    }
-}
-
-sub appendCaseStatus
-{
-    my $msg = shift;
-    open (my $CS, ">>", "$config{'CASEROOT'}/CaseStatus") or warn "cannot open $config{'CASEROOT'}/CaseStatus, $!";
-    print $CS "$msg\n";
-    close $CS;
-}
-
-# check to see if we need to do resubmission from this particular job,
-sub resubmitCheck()
-{
-#
-# Mira requires special logic
-#
-    if(($config{DOUT_S} eq 'FALSE' && $config{RESUBMIT} > 0) || ($config{MACH} eq "mira" && $config{DOUT_S} eq "TRUE")) {
-	my $submitscript = "$config{'CASEROOT'}/case.submit";
-	my $cwd = getcwd;
-	chdir $config{'CASEROOT'};
-	my $resubmitcommand = "$submitscript -resubmit -scriptname case.";
-
-	if(defined $config{TESTCASE} && $config{TESTCASE} eq "ERR"){
-	    $resubmitcommand .= "test";
-	}else{
-	    $resubmitcommand .= "run";
-	}
-	$logger->debug( "running resubmit check $resubmitcommand");
-	system($resubmitcommand);
-        if($?){ $logger->warn ("could not run ACME submit script, $! $?");}
-	chdir $cwd;
-    }
-}
-
-sub DoDataAssimilation
-{
-    my ($rundir, $caseroot, $da, $da_script, $lid, $logdir) = @_;
-
-    return unless($da eq "TRUE");
-
-    system(" $da_script 1> da.log.$lid 2>&1");
-    sleep(1);
-    $logger->logdie ("$da_script failed") unless ($? == 0);
-    chdir $rundir;
-    disposeLog($rundir,$caseroot, 'da' ,$lid,$logdir);
-    chdir $caseroot;
-}
+    # create the timing directories, optionally cleaning them if needed.
+    try:
+        os.stat(rundir)
```

Comment:
Don't use exceptions to handle normal code flow. I think you want this:

    if not os.path.isdir(rundir):
        os.mkdir(rundir)